File: C#_LeftToDo/Menus/Menu.cs
using System;
using LeftToDo.Lists;
using LeftToDo.Tasks;

namespace LeftToDo.Menus
{
    abstract class Menu // Menu displays and handles out- / input
    {
        // Displays main menu 
        internal static void ShowMainMenu(TaskList list)
        {
            Console.WriteLine("\tHUVUDMENY\n[1] Visa Att-göra uppgifter\n[2] Visa Arkiverade uppgifter\n[0] Avsluta");
            ShowIntro();
            var menu = Console.ReadLine();
            switch (menu)
            {
                case "1":
                    list.ShowLeftToDo(list.ToDoList);
                    ShowToDoMenu(list);
                    break;
                case "2":
                    list.ShowArchive(list.Archive);
                    ShowArchiveMenu(list);
                    break;
                case "0":
                    ShowFarewell();
                    break;
                default:
                    ShowErrorMsg();
                    ShowMainMenu(list);
                    break;
            }
        }

        // Displays instructions to user
        private static void ShowIntro()
        {
            Console.WriteLine("\nVälj siffra inom [] följt av enter.\n");
        }

        // Displays submenu
        private static void ShowToDoMenu(TaskList list)
        {
            Console.WriteLine(
                "\n\tUNDERMENY: Vad vill du göra?\n" +
                "[1] Lägg till uppgift\n" +
                "[2] Markera / avmarkera uppgift\n" +
                "[3] Arkivera utförda uppgifter\n" +
                "[0] Tillbaka till HUVUDMENY"
            );
            ShowIntro();
            var menu = Console.ReadLine();
            switch (menu)
            {
                case "1":
                    ShowTaskMenu(list);
                    break;
                case "2":
                    list.FindTaskToMark(list);
                    ShowToDoMenu(list);
                    break;
                case "3":
                    list.ArchiveTask();
                    list.ShowLeftToDo(list.ToDoList);
                    ShowToDoMenu(list);
                    break;
                case "0":
                    ShowMainMenu(list);
                    break;
                default:
                    ShowErrorMsg();
                    break;
            }
        }

        // Display task menu
        private static void ShowTaskMenu(TaskList list)
        {
            list.ShowLeftToDo(list.ToDoList);
            Console.WriteLine(
                "\n\tUPPGIFTSMENY: Välj typ av uppgift:\n" +
                "[1] Enkel uppgift\n" +
                "[2] Checklista\n" +
                "[0] Tillbaka till HUVUDMENY"
            );
            ShowIntro();
            var menu = Console.ReadLine();
            switch (menu)
            {
                case "1":
                    var simpelTask = new SimpleTask("");
                    simpelTask.Create();
                    list.AddToDoTask(simpelTask);
                    list.ShowLeftToDo(list.ToDoList);
                    ShowToDoMenu(list);
                    break;
                case "2":
                    var checklist = new Checklist("");
                    checklist.Create();
                    list.AddToDoTask(checklist);
                    list.ShowLeftToDo(list.ToDoList);
                    ShowToDoMenu(list);
                    break;
                case "0":
                    ShowMainMenu(list);
                    break;
                default:
                    ShowErrorMsg();
                    break;
            }
        }

        // Displays option for user to return to main menu
        private static void ShowArchiveMenu(TaskList list)
        {
            Console.WriteLine("\n[0] Tillbaka till HUVUDMENY\n");
            var input = Console.ReadLine();
            if (input == "0")
            {
                ShowMainMenu(list);
            }
        }

        // Displays farewell message
        private static void ShowFarewell()
        {
            Console.WriteLine("\t\tTack för besöket och välkommen åter!\n\n");
            return;
        }

        // Displays error message
        private static void ShowErrorMsg()
        {
            Console.WriteLine("\t\tOgiltligt val. Vargod försök igen!\n\n");
        }
    }
}

File: Go_leftToDo/menu/menu.go
// Package menu displays and handles out- / input
package menu

import (
	"fmt"

	"github.com/WiviWonderWoman/DevUx/Go/lists"
	"github.com/WiviWonderWoman/DevUx/Go/tasks"
)

// Displays main menu
func ShowMainMenu(list lists.TaskList) {
	fmt.Printf("\n\tHUVUDMENY\n[1] Visa Att-göra uppgifter\n[2] Visa Arkiverade uppgifter\n[0] Avsluta\n")
	showIntro()
	var input string
	fmt.Scanln(&input)
	switch input {
	case "1":
		list.ShowLeftToDo()
		showToDoMenu(list)
	case "2":
		list.ShowArchive()
		showArchiveMenu(list)
	case "0":
		showFarewell()
	default:
		showErrorMsg()
		ShowMainMenu(list)
	}
}

// Displays instructions to user
func showIntro() {
	fmt.Printf("\nVälj siffra inom [] följt av enter.\n")
}

// Displays submenu
func showToDoMenu(list lists.TaskList) {
	fmt.Printf(
		"\n\tUNDERMENY: Vad vill du göra?\n" +
			"[1] Lägg till uppgift\n" +
			"[2] Markera / avmarkera uppgift\n" +
			"[3] Arkivera utförda uppgifter\n" +
			"[0] Tillbaka till HUVUDMENY\n",
	)
	showIntro()
	var input string
	fmt.Scanln(&input)
	switch input {
	case "1":
		showTaskMenu(list)
	case "2":
		list.FindTaskToMark()
		showToDoMenu(list)
	case "3":
		list.ArchiveTask()
		list.ShowLeftToDo()
		showToDoMenu(list)
	case "0":
		ShowMainMenu(list)
	default:
		showErrorMsg()
		showToDoMenu(list)
	}
}

// Display task menu
func showTaskMenu(list lists.TaskList) {
	list.ShowLeftToDo()
	fmt.Printf(
		"\n\tUPPGIFTSMENY. Välj typ av uppgift:\n" +
			"[1] Enkel uppgift\n" +
			"[2] Checklista\n" +
			"[0] Tillbaka till HUVUDMENY\n",
	)
	showIntro()
	var input string
	fmt.Scanln(&input)
	switch input {
	case "1":
		simple := tasks.SimpleTask{}.Create()
		list.AddToDoTask(simple.Task)
		list.ShowLeftToDo()
		showToDoMenu(list)
	case "2":
		checklist := tasks.ChecklistTask{}.Create()
		list.AddToDoTask(checklist.Task)
		list.ShowLeftToDo()
		showToDoMenu(list)
	case "0":
		ShowMainMenu(list)
	default:
		showErrorMsg()
		showTaskMenu(list)
	}
}

// Displays option for user to return to main menu
func showArchiveMenu(list lists.TaskList) {
	fmt.Printf("\n[0] Tillbaka till HUVUDMENY\n")
	var input string
	fmt.Scanln(&input)
	if input == "0" {
		ShowMainMenu(list)
	} else {
		showErrorMsg()
		showArchiveMenu(list)
	}
}

// Displays farewell message
func showFarewell() {
	fmt.Printf("\n\t\tTack för besöket och välkommen åter!\n\n")
}

// Displays error message
func showErrorMsg() {
	fmt.Printf("\n\t\tOgiltligt val. Vargod försök igen!\n\n")
}
